server/src/main/java/org/elasticsearch/index/engine/LiveVersionMap.java

```diff
@@ -142,6 +158,11 @@ Maps invalidateOldMap() {
             return new Maps(current, VersionLookup.EMPTY, previousMapsNeededSafeAccess);
```

Comment:
Can we rename this method to `invalidateOldMapForAssertion` or something like that? It clearly should never be called for production code.